tools/msgdis.py

```diff
@@ -0,0 +1,442 @@
+    fra_message_entry_table = list(zip(ids,as_word_list(baserom[fra_message_entry_table_addr:staff_message_entry_table_addr])))
+
+    for entry in nes_message_entry_table:
+        if entry[0] != 0xFFFC:
+            combined_message_entry_table.append((*entry, dict(ger_message_entry_table)[entry[0]], dict(fra_message_entry_table)[entry[0]]))
+        else:
+            combined_message_entry_table.append((*entry, None, None))
+
+    staff_message_entry_table = as_message_table_entry(baserom[staff_message_entry_table_addr:staff_message_entry_table_addr_end])
+
+# ===================================================
+#   Run
+# ===================================================
+
+### TODO this is all awful
+def cvt(m):
+    return m.group(0).replace("\"\"","")
+
+doubles = re.compile(r"(?<!\\)(\"\")")
+def fixup_message(message):
+    return re.sub(doubles, cvt, ("\"" + message.replace("\n","\\n\"\n\"") + "\"")).replace("\n ","\n").replace("BOX_BREAK\"","\nBOX_BREAK\n\"").replace("BOX_BREAK ","\nBOX_BREAK\n").strip()
+###
+
+def dump_all_text():
+    # text id, ypos, type, nes, ger, fra
+    messages = []
+    for i,entry in enumerate(combined_message_entry_table,0):
+        if not (entry[0] == 0xFFFD or entry[0] == 0xFFFF):
+            next_entry = combined_message_entry_table[i+1]
+
+            nes_offset = segmented_to_physical(entry[3])
+            nes_length = next_entry[3] - entry[3]
+            nes_text = ""
+            with open("baserom/nes_message_data_static","rb") as infile:
+                infile.seek(nes_offset)
+                nes_text = fixup_message(decode(infile.read(nes_length), entry[1]).replace("\x00","",-1))
+
+            ger_text = ""
+            fra_text = ""
+            if entry[0] != 0xFFFC:
+                if next_entry[0] == 0xFFFC:
+                    next_entry = combined_message_entry_table[i+2]
+                ger_offset = segmented_to_physical(entry[4])
+                ger_length = next_entry[4] - entry[4]
+                with open("baserom/ger_message_data_static","rb") as infile:
+                    infile.seek(ger_offset)
+                    ger_text = fixup_message(decode(infile.read(ger_length), entry[1]).replace("\x00","",-1))
+                
+                fra_offset = segmented_to_physical(entry[5])
+                fra_length = next_entry[5] - entry[5]
+                with open("baserom/fra_message_data_static","rb") as infile:
+                    infile.seek(fra_offset)
+                    fra_text = fixup_message(decode(infile.read(fra_length), entry[1]).replace("\x00","",-1))
+
+            messages.append((entry[0], entry[1], entry[2], nes_text, ger_text, fra_text))
+        elif entry[0] == 0xFFFD:
+            messages.append((entry[0], entry[1], entry[2], "\"\"", "\"\"", "\"\""))
+        elif entry[0] == 0xFFFF:
+            messages.append((entry[0], entry[1], entry[2], "///END///", "///END///", "///END///"))
+        else:
+            assert False , "What happened?"
```

Comment:
instead of
```py
if not 0xFFFD and not 0xFFFF
 ...
elif 0xFFFD
 ...
elif 0xFFFF
 ...
else
 fail
```
I think a better (less confusing) way to write this would be
```py
if 0xFFFD
 ...
elif 0xFFFF
 ...
else
 ... # not 0xFFFD and not 0xFFFF
```